package/boot/uboot-mvebu/patches/139-arm-mvebu-Espressobin-Setup-MTD-partitions-when-boot.patch

```diff
@@ -0,0 +1,143 @@
+From 4822bb409f267253a31eb01c1256f0fb8a2253bf Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Pali=20Roh=C3=A1r?= <pali@kernel.org>
+Date: Wed, 19 Aug 2020 16:24:17 +0200
+Subject: [PATCH] arm: mvebu: Espressobin: Setup MTD partitions when booting
+ kernel
```

Comment:
What's the use for this patch? The space left won't fit kernel and whole flash is intended for bootloader only anyway.

Reply:
With the mtd partitions exposed to linux you can use fw_setenv/fw_printenv from linux.

Reply:
Flash is not only for firmware/bootloader/u-boot, but also for u-boot environment. This patch ensures that Linux kernel would see correct MTD layout based on how was u-boot compiled (what is CONFIG_ENV_OFFSET).

Reply:
Kernel does not need to see that, it will map the whole flash as single mtd device anyway and fw_setenv/fw_printenv can run with offsets, just like it's already done https://github.com/openwrt/openwrt/blob/22caf30a65aaa8a880b1a935bed035ea0826e746/package/boot/uboot-envtools/files/mvebu#L29. Anyway better approach would be specifying `mtdparts`.

Reply:
`0x3f0000` is not fixed number, but compile time option which user can change to whatever want. There is just default value. In kernel there was already MTD layout specified in DTS file, but was removed due to this fact and also due fact that Marvell distributed more uboot versions with different offset value.

So the main issue that current running uboot must somehow inform kernel what is the correct offset. And this information is stored in DTS/FDT file.

U-Boot already provides functions to automatically set MTD layout in FDT, but somehow it is broken for EspressoBin binary (there is discussion on u-boot list about it). I think that if somebody fixes that bug, U-Boot function could be replaced by this code.

Reply:
Yeah, this config symbol is exposed so environment could be at any place. Ok, that's plausible explanation, I'm fine with keeping this patch.